pkg/provider/azure/keyvault/keyvault.go

```diff
@@ -491,26 +493,27 @@ type tokenProvider struct {
 	accessToken string
 }
 
-type tokenProviderFunc func(ctx context.Context, token, clientID, tenantID string) (adal.OAuthTokenProvider, error)
+type tokenProviderFunc func(ctx context.Context, token, clientID, tenantID, aadEndpoint, kvResource string) (adal.OAuthTokenProvider, error)
 
-func newTokenProvider(ctx context.Context, token, clientID, tenantID string) (adal.OAuthTokenProvider, error) {
+func newTokenProvider(ctx context.Context, token, clientID, tenantID, aadEndpoint, kvResource string) (adal.OAuthTokenProvider, error) {
 	// exchange token with Azure AccessToken
 	cred, err := confidential.NewCredFromAssertion(token)
 	if err != nil {
 		return nil, err
 	}
-
-	// AZURE_AUTHORITY_HOST
-
 	cClient, err := confidential.New(clientID, cred, confidential.WithAuthority(
-		fmt.Sprintf("https://login.microsoftonline.com/%s/oauth2/token", tenantID),
+		fmt.Sprintf("%s%s/oauth2/token", aadEndpoint, tenantID),
 	))
 	if err != nil {
 		return nil, err
 	}
-
+	scope := kvResource
+	// .default needs to be added to the scope
+	if !strings.Contains(kvResource, ".default") {
+		scope = fmt.Sprintf("%s/.default", kvResource)
+	}
```

Comment:
I couldn't find docs around the why. Taken from CSISS: https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/8a9e262d01f92829c3e888a9c9f8ea5db1b67a0a/pkg/auth/auth.go#L151-L153